File: Backend/src/main/java/com/copernic/backend/Backend/entity/enums/Estat.java
package com.copernic.backend.Backend.entity.enums;

/**
 * Enum que representa els diferents estats d'una recompensa dins del sistema.
 * S'utilitza per controlar el flux i disponibilitat de les recompenses en el procés de reserva i recollida.
 */
public enum Estat {

    /**
     * La recompensa està disponible per ser reservada.
     */
    DISPONIBLES,

    /**
     * La recompensa ha estat reservada per un usuari però encara no assignada.
     */
    RESERVADES,

    /**
     * La recompensa ha estat assignada a un usuari i està pendent de recollida.
     */
    ASSIGNADES,

    /**
     * La recompensa ja ha estat recollida per l'usuari.
     */
    RECOLLIDA,

    /**
     * La recompensa està pendent de recollir però ja ha estat reservada i gestionada.
     */
    PER_RECOLLIR
}

File: Frontend/app/src/main/java/cat/copernic/amayo/frontend/recompensaManagment/model/Estat.kt
package cat.copernic.amayo.frontend.recompensaManagment.model

/**
 * Enum que representa els possibles estats d'una recompensa.
 *
 * Defineix les diferents fases o situacions en què pot trobar-se una recompensa
 * dins del sistema de gestió de reserves i bescanvis.
 */
enum class Estat {

    /** La recompensa està disponible per ser reservada. */
    DISPONIBLES,

    /** La recompensa ha estat reservada per un usuari. */
    RESERVADES,

    /** La recompensa ha estat assignada a un usuari. */
    ASSIGNADES,

    /** La recompensa ha estat recollida per l'usuari. */
    RECOLLIDA,

    /** La recompensa està pendent de ser recollida. */
    PER_RECOLLIR
}